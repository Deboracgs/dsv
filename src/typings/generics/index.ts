export * from "./redux"
export type Nullable<T> = T | null;
export interface IGeo {
    lat: string;
    lng: string;
}