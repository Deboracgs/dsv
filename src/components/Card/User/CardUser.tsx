import React from "react";
import { Card, CardContent } from "@mui/material";
import { IUser } from "./../../../typings";
import { CardUserStyled, UserAddressStyled, UserCompanyStyled, UserInfoStyled } from "./CardUser.styled";
import {Button} from "./../../index";

export interface CardUserProps {
  id: string;
  onRemove?: (value: string) => void;
  onRestore?: (value: string) => void;
  user: IUser;
}

const CardUser = ({
  id,
  onRemove,
  onRestore,
  user
}: CardUserProps): React.ReactElement => {
  const onClick = () => {
    if(user.deleted){
      if (typeof onRestore === "function") onRestore(id);
    }else{
      if (typeof onRemove === "function") onRemove(id);
    }
    
  };

  const text = user.deleted ? "Restore" : "Remove";
  return (
    <CardUserStyled data-testid={`card-user-${id}`}>
      <Card>
        <CardContent>
          <UserInfoStyled variant="h6">{user.username} - {user.age} </UserInfoStyled>
          <UserCompanyStyled variant="subtitle1">{user.companyName}</UserCompanyStyled>
          <UserAddressStyled variant="subtitle2">{user.address.street}, {user.address.suite}, {user.address.city} - {user.address.zipcode}</UserAddressStyled>
          <Button id={`${text}-user${id}`} label={`${user.deleted ? "Restore" : "Remove"} User`} variant="outlined" onClick={() => onClick()} />
        </CardContent>
      </Card>
    </CardUserStyled>
    
  );
};

CardUser.defaultProps = {
  onRemove: undefined,
  onRestore: undefined
};
export default CardUser;
