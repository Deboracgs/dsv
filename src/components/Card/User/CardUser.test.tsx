/* eslint-disable jest/no-mocks-import */
import { fireEvent, render } from "@testing-library/react";
import users from "./../../../__mocks__/users";
import WrapperMock from "../../../utils/wrapperMock";
import CardUser, { CardUserProps } from "./CardUser";
import { IUser } from "../../../typings";

describe("Card User Component", () => {
  const defaultProps: CardUserProps = {
    id: "test",
    user: users[0] as unknown as IUser,
    onRemove: jest.fn(),
    onRestore: jest.fn()
  };

  const rendered = (props: CardUserProps) =>
    render(
      <WrapperMock>
        <CardUser {...props} />
      </WrapperMock>
    );

  it("render", () => {
    const { getByTestId, asFragment } = rendered(defaultProps);

    const element = getByTestId(`card-user-${defaultProps.id}`);

    expect(element).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it("onRemove", () => {
    const onClickSpy = jest.spyOn(defaultProps, "onRemove");
    const { getByTestId } = rendered(defaultProps);

    const btnCommon = getByTestId(`button-Remove-usertest`);
    fireEvent.click(btnCommon);

    expect(onClickSpy).toHaveBeenCalled();
  });

  it("onRestore", () => {
    const onClickSpy = jest.spyOn(defaultProps, "onRestore");
    const { getByTestId } = rendered({...defaultProps, user: {...defaultProps.user, deleted: true}});

    const btnCommon = getByTestId(`button-Restore-usertest`);
    fireEvent.click(btnCommon);

    expect(onClickSpy).toHaveBeenCalled();
  });
});
