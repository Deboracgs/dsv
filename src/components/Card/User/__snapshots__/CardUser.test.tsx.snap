// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card User Component render 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ hYcics"
    data-testid="card-user-test"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 sc-gsnTZi llGvie css-2ulfj5-MuiTypography-root"
        >
          Bret - 18 
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 sc-dkzDqf ktnddZ css-10wpov9-MuiTypography-root"
        />
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 sc-hKMtZM jpMeVU css-ati914-MuiTypography-root"
        >
          Kulas Light, Apt. 556, Gwenborough - 92998-3874
        </h6>
        <button
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          data-testid="button-Remove-usertest"
          id="button-Remove-usertest"
          tabindex="0"
          type="button"
        >
          Remove User
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
